libpromises/evalfunction.c

```diff
@@ -1188,6 +1188,95 @@ static FnCallResult FnCallGetGid(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const P
 #endif
 }
 
+/*********************************************************************/ 
+
+static FnCallResult FnCallUserInGroup(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+{
+    assert(fp != NULL);
+#ifdef _WIN32
+    Log(LOG_LEVEL_ERR, "Function '%s' is POSIX specific", fp->name);
+    return FnFailure();  
+#else
+    assert(finalargs != NULL);
```

Comment:
This could be put before the ifdef. Asserts are testing our assumptions as programmers and looking for programmer errors / mistakes. This function should always be called with `finalargs != NULL`, if not, that's an error, I think, even on Windows.

It's also quite nice to have these asserts at the very top of the function - a bit nicer to see what is the acceptable way of calling this function.